presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -560,61 +566,55 @@ private static Split createRemoteSplitFor(TaskId taskId, URI remoteSourceTaskLoc
 
     private void updateTaskStatus(TaskId taskId, TaskStatus taskStatus)
     {
-        try {
-            StageExecutionState stageExecutionState = getState();
-            if (stageExecutionState.isDone()) {
-                return;
-            }
+        StageExecutionState stageExecutionState = getState();
```

Comment:
I fail to see why the try-catch was removed and this function has changed ?

Reply:
It wasn't a try/catch it was a try/finally (i.e. no exceptions were being caught).  the finally block adds a call to checkAllTasksFinal(), but the call isn't be necessary since we already have listeners for the state machine changes that would make checkAllTasksFinal actually do anything.  this change is just removing that call.